### PR TITLE
DBZ-252 Antlr parser rebased and with review adjustments

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-252

Supersedes #534.

@jpechane Quite a few of my original review comments went unnoticed. I've addressed the most important ones here. Also going to add some more things later on, then it should be alright to be merged. Main issues were that it used Antlr by default and the old one couldn't be used at all.

### CODE REVIEW

File: debezium-ddl-parser/src/main/java/io/debezium/antlr/ProxyParseTreeListenerUtil.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+/*
```
Comment (gunnarmorling) [inline debezium-ddl-parser/src/main/java/io/debezium/antlr/ProxyParseTreeListenerUtil.java]: Hey @kucharo2, why is that licensed under BSD 3-clause?

**Comment by gunnarmorling** [comment]
And merged. Thanks a ton again, @kucharo2; really excited about having this in!

File: debezium-ddl-parser/src/main/java/io/debezium/antlr/ProxyParseTreeListenerUtil.java (hunk 1 above)
Comment (kucharo2) [reply debezium-ddl-parser/src/main/java/io/debezium/antlr/ProxyParseTreeListenerUtil.java]: I was not sure about the licence header in this one. The implementation is taken from ANTLR repo, and they are using BSD license (the same is with CaseChangingCharStream).

**Comment by kucharo2** [comment]
@gunnarmorling Happy to hear that, I hope that there will be only positive feedbacks for this.

**Pull request closed without merging**